### PR TITLE
SQL: handle SQL not being available in a more graceful way

### LINKED ISSUE #41279: SQL: handle SQL not being available on server in a more graceful way
Using the JDBC driver on an OSS version of Elasticsearch generates a rather un-friendly error message. The message itself is technically correct, but is not a graceful way of letting the user know that the node it's trying to connect to doesn't have SQL installed.

This issue is very similar to https://github.com/elastic/elasticsearch/issues/30009.

```
Exception in thread "main" org.elasticsearch.xpack.sql.client.ClientException: Cannot POST address http://localhost:9200/_sql?error_trace (Can't parse error from Elasticsearch [Expected [error] to be an object but was [VALUE_STRING][Incorrect HTTP method for uri [/_sql?error_trace] and method [POST], allowed: [PUT, DELETE, GET, HEAD]]] at [line 1 col 10]. Response:
{"error":"Incorrect HTTP method for uri [/_sql?error_trace] and method [POST], allowed: [PUT, DELETE, GET, HEAD]","status":405})
	at org.elasticsearch.xpack.sql.client.JreHttpUrlConnection.request(JreHttpUrlConnection.java:163)
	at org.elasticsearch.xpack.sql.client.HttpClient.lambda$post$1(HttpClient.java:103)
	at org.elasticsearch.xpack.sql.client.JreHttpUrlConnection.http(JreHttpUrlConnection.java:65)
	at org.elasticsearch.xpack.sql.client.HttpClient.lambda$post$2(HttpClient.java:102)
	at java.security.AccessController.doPrivileged(Native Method)
	at org.elasticsearch.xpack.sql.client.HttpClient.post(HttpClient.java:101)
	at org.elasticsearch.xpack.sql.client.HttpClient.query(HttpClient.java:78)
	at org.elasticsearch.xpack.sql.jdbc.JdbcHttpClient.query(JdbcHttpClient.java:60)
	at org.elasticsearch.xpack.sql.jdbc.JdbcStatement.initResultSet(JdbcStatement.java:160)
	at org.elasticsearch.xpack.sql.jdbc.JdbcPreparedStatement.executeQuery(JdbcPreparedStatement.java:60)
	at com.foo.Main.main(Main.java:34)
Caused by: java.io.IOException: Can't parse error from Elasticsearch [Expected [error] to be an object but was [VALUE_STRING][Incorrect HTTP method for uri [/_sql?error_trace] and method [POST], allowed: [PUT, DELETE, GET, HEAD]]] at [line 1 col 10]. Response:
{"error":"Incorrect HTTP method for uri [/_sql?error_trace] and method [POST], allowed: [PUT, DELETE, GET, HEAD]","status":405}
	at org.elasticsearch.xpack.sql.client.RemoteFailure.parseFromResponse(RemoteFailure.java:75)
	at org.elasticsearch.xpack.sql.client.JreHttpUrlConnection.parserError(JreHttpUrlConnection.java:174)
	at org.elasticsearch.xpack.sql.client.JreHttpUrlConnection.request(JreHttpUrlConnection.java:161)
	... 10 more
Caused by: java.io.IOException: Expected [error] to be an object but was [VALUE_STRING][Incorrect HTTP method for uri [/_sql?error_trace] and method [POST], allowed: [PUT, DELETE, GET, HEAD]]
	at org.elasticsearch.xpack.sql.client.RemoteFailure.parseResponseTopLevel(RemoteFailure.java:159)
	at org.elasticsearch.xpack.sql.client.RemoteFailure.parseFromResponse(RemoteFailure.java:71)
	... 12 more
```

### PR DESCRIPTION
**Pull request by astefan**
When https://github.com/elastic/elasticsearch/pull/36149 was merged, the default requests the JDBC driver was making changed slightly (the `mode` parameter was not a parameter anymore) and, also, the rest requests the ES server was receiving were handled slightly differently in the RestController.
So differently, that the error message being generated when a SQL request was made to a cluster that didn't have SQL installed (or disabled) was different.

This PR changes this error message handling and, also, adds a much needed integration test to handles this scenario. The test is a separate QA project which, in Gradle configuration, has SQL disabled.

Fixes https://github.com/elastic/elasticsearch/issues/41279.

### CODE REVIEW

File: x-pack/plugin/sql/qa/src/main/java/org/elasticsearch/xpack/sql/qa/jdbc/JdbcNoSqlTestCase.java [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+package org.elasticsearch.xpack.sql.qa.jdbc;
+
+import java.sql.Connection;
+import java.sql.SQLException;
+
+public class JdbcNoSqlTestCase extends JdbcIntegrationTestCase {
+    public void testJdbcExceptionMessage() throws SQLException {
+        try (Connection c = esJdbc()) {
+            SQLException e = expectThrows(SQLException.class, () -> c.prepareStatement("SELECT * FROM bla").executeQuery());
+            assertTrue(e.getMessage().startsWith("X-Pack/SQL do not seem to be available on the Elasticsearch node using the access path"));
```
Comment (matriv) [inline x-pack/plugin/sql/qa/src/main/java/org/elasticsearch/xpack/sql/qa/jdbc/JdbcNoSqlTestCase.java]: do -> does

File: x-pack/plugin/sql/sql-client/src/main/java/org/elasticsearch/xpack/sql/client/JreHttpUrlConnection.java [hunk 2]
```diff
@@ -48,8 +48,10 @@
      * error.
      */
     public static final String SQL_STATE_BAD_SERVER = "bad_server";
-    private static final String SQL_NOT_AVAILABLE_ERROR_MESSAGE = "request [" + SQL_QUERY_REST_ENDPOINT
+    private static final String SQL_NOT_AVAILABLE_ERROR_MESSAGE_PARAMETER = "request [" + SQL_QUERY_REST_ENDPOINT
             + "] contains unrecognized parameter: [mode]";
+    private static final String SQL_NOT_AVAILABLE_ERROR_MESSAGE_HTTPMETHOD = "Incorrect HTTP method for uri [" + SQL_QUERY_REST_ENDPOINT
```
Comment (matriv) [inline x-pack/plugin/sql/sql-client/src/main/java/org/elasticsearch/xpack/sql/client/JreHttpUrlConnection.java]: Is this case tested somewhere else?

**Review by matriv — COMMENTED** [review]
Seems good, left a question.

File: x-pack/plugin/sql/sql-client/src/main/java/org/elasticsearch/xpack/sql/client/JreHttpUrlConnection.java (hunk 2 above)
Comment (astefan) [reply x-pack/plugin/sql/sql-client/src/main/java/org/elasticsearch/xpack/sql/client/JreHttpUrlConnection.java]: The special case of sql not being installed/disabled had no tests before this one.

**Comment by astefan** [comment]
@elastic/es-core-infra is there any other solution that you might think of for this integration test to be possible?
Please, note that I created a new qa sub-project where I am disabling SQL from Gradle config, so that I can test the JDBC driver error message when SQL is not installed or disabled. ESRestTestCase nor ESIntegTestCase allow me, at the same time, to disable SQL from code (since from Gradle I would disable SQL for the entire project) AND get a useful hostname and port number to use for the JDBC driver connection.

**Review by matriv — APPROVED** [review]
LGTM

**Pull request merged**